transaction/executer_test.go

```diff
@@ -233,363 +232,90 @@ func Test_Executer_TransactionIDGiven_NoReplay(t *testing.T) {
 	}
 }
 
-func Test_Executer_TransactionResult_Byte(t *testing.T) {
-	config := DefaultExecuterConfig()
-	newExecuter, err := NewExecuter(config)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
-	}
-
-	var trialOutput []byte
-	var replayInput []byte
-
-	replay := func(context context.Context, v interface{}) error {
-		replayInput = []byte(v.(string))
-		return nil
-	}
-	trial := func(context context.Context) (interface{}, error) {
-		trialOutput = []byte("hello world")
-		return trialOutput, nil
-	}
-
-	var ctx context.Context
-	var executeConfig ExecuteConfig
-	{
-		ctx = context.Background()
-		ctx = transactionid.NewContext(ctx, "test-transaction-id")
-
-		executeConfig = newExecuter.ExecuteConfig()
-		executeConfig.Replay = replay
-		executeConfig.Trial = trial
-		executeConfig.TrialID = "test-trial-ID"
-	}
-
-	{
-		err := newExecuter.Execute(ctx, executeConfig)
-		if err != nil {
-			t.Fatal("expected", nil, "got", err)
-		}
-
-		if string(trialOutput) != "hello world" {
-			t.Fatal("expected", "hello world", "got", string(trialOutput))
-		}
-		if string(replayInput) != "" {
-			t.Fatal("expected", "", "got", string(replayInput))
-		}
-	}
-
-	{
-		err := newExecuter.Execute(ctx, executeConfig)
-		if err != nil {
-			t.Fatal("expected", nil, "got", err)
-		}
-
-		if string(trialOutput) != "hello world" {
-			t.Fatal("expected", "hello world", "got", string(trialOutput))
-		}
-		if string(replayInput) != "hello world" {
-			t.Fatal("expected", "hello world", "got", string(replayInput))
-		}
-	}
-}
-
-func Test_Executer_TransactionResult_Float64(t *testing.T) {
-	config := DefaultExecuterConfig()
-	newExecuter, err := NewExecuter(config)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
-	}
-
-	var trialOutput float64
-	var replayInput float64
-
-	replay := func(context context.Context, v interface{}) error {
-		var err error
-		replayInput, err = strconv.ParseFloat(v.(string), 64)
-		if err != nil {
-			t.Fatal("expected", nil, "got", err)
-		}
-		return nil
-	}
-	trial := func(context context.Context) (interface{}, error) {
-		trialOutput = 4.3
-		return trialOutput, nil
-	}
-
-	var ctx context.Context
-	var executeConfig ExecuteConfig
-	{
-		ctx = context.Background()
-		ctx = transactionid.NewContext(ctx, "test-transaction-id")
-
-		executeConfig = newExecuter.ExecuteConfig()
-		executeConfig.Replay = replay
-		executeConfig.Trial = trial
-		executeConfig.TrialID = "test-trial-ID"
-	}
-
-	{
-		err := newExecuter.Execute(ctx, executeConfig)
+func Test_Executer_TransactionResult(t *testing.T) {
+	tests := []struct {
+		TrialOutput     interface{}
+		WantReplayInput interface{}
+	}{
+		{ // 0
```

Comment:
I would like to see comments in a separate line and a bit more elaborative. This is pretty useless to me personally.